src/hotspot/share/runtime/thread.cpp

```diff
@@ -1613,13 +1592,14 @@ void JavaThread::remove_monitor_chunk(MonitorChunk* chunk) {
   }
 }
 
-// JVM support.
 
+// Asynchronous exceptions support
+//
 // Note: this function shouldn't block if it's called in
 // _thread_in_native_trans state (such as from
 // check_special_condition_for_native_trans()).
-void JavaThread::check_and_handle_async_exceptions(bool check_unsafe_error) {
```

Comment:
Hi David,

> I'm not clear how you got rid of the check_unsafe_error. They are a special kind of async exception and the scope of checking for them was supposed to be very narrow.

The only time we call check_and_handle_async_exceptions() with check_unsafe_error=false is in check_special_condition_for_native_trans(), in the transition from native to Java, otherwise that argument is always true. I only moved the check to has_async_exception_condition() so for that particular case we only call check_and_handle_async_exceptions() for non _async_unsafe_access_error cases. 
That boolean is actually redundant today because in  check_special_condition_for_native_trans() we call has_async_exception() first which checks the _suspend_flag bit instead of _special_runtime_exit_condition, and that already only gets set for non _async_unsafe_access_error cases. Then in check_and_handle_async_exceptions() we always check for non _async_unsafe_access_error cases first.

Thanks,
Patricio